src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private final Operation operation;
+    private final String fruit;
+    private final int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        // Метод для конвертації коду в Operation
+        public static Operation fromCode(String code) {
+            for (Operation operation : values()) {
+                if (operation.getCode().equalsIgnoreCase(code)) {
+                    return operation;
+                }
+            }
+            throw new IllegalArgumentException("No enum constant with code " + code);
+        }
+    }
+}
```

Comment:
Consider following the encapsulation principle. The members of the `FruitTransaction` class are currently all private and final, which is good. However, the `Operation` enum is public and could be moved to its own file if it's used outside of `FruitTransaction`, or made private if it's only used within this class.